test/listusers.test.js

```diff
@@ -0,0 +1,56 @@
+const request = require('supertest');
+const app = require('../app');
+const { factory } = require('factory-girl');
+
+let user = null;
+
+beforeAll(async () => {
+  const userFactoryGirlResponse = await factory.build('User');
+  user = userFactoryGirlResponse.dataValues;
+});
+
+describe('GET /users', () => {
+  test('token missing', async () => {
+    await request(app)
+      .post('/users')
+      .send(user)
+      .set('Accept', 'application/json');
+    await request(app)
+      .post('/users/sessions')
+      .send(user)
+      .set('Accept', 'application/json');
+    const response = await request(app).get('/users');
```

Comment:
1. Move user creation and authentication into a helper in order to not repeat yourself in each test.

2. When you have two or more promises that can be executed parallel do it using `Promise.all(promises)`. 

3. Do you need to call the controller to populate your database for tests? Shouldn't be better to directly call database?